docs/concurrency.md

```diff
@@ -0,0 +1,47 @@
+## Concurrency control primitives
+
+The "ConcurrencyControl" CRD is used for higher-level systems to control concurrency of PipelineRuns.
+To use concurrency controls, first create an instance of a ConcurrencyControl, for example:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: ConcurrencyControl
+metadata:
+  name: my-concurrency-control
+spec:
+  params:
+  - name: param-1
+  - name: param-2
```

Comment:
Those would be "populated" by the `PipelineRun` params ?
What happens if there is a mismatch between `PipelineRun`'s params and that `ConcurrencyControl` params?